.travis.yml

```diff
@@ -46,11 +46,14 @@ deploy:
 jobs:
   include:
     - stage: Build & Bundle
-      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="robot_ws"
+      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="robot_ws" UPLOAD_SOURCES="false"
     - stage: Build & Bundle
-      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="simulation_ws"
+      env: ROS_DISTRO=dashing ROS_VERSION=2 GAZEBO_VERSION=9 WORKSPACES="simulation_ws" UPLOAD_SOURCES="false"
+    - stage: Upload sources
+      script: . .ros_ci/add_tag.sh && .ros_ci/prepare_sources_to_upload.sh && set +e
```

Comment:
Why is `. .ros_ci/add_tag.sh` still here?

Reply:
It exports SA_VERSION which we need for `before_deploy` and `deploy`.

Reply:
"Upload Sources" stage doesn't have `before_deploy` and `deploy`. Are you saying this export from "Upload Sources" stage needs to be propagated to the `before_deploy` and `deploy` of the "Deploy" stage?

Reply:
It doesn't have an override for those, so it will use the `before_deploy` and `deploy` steps that are defined globally.